doc/reference/dune-project/pin.rst

```diff
@@ -0,0 +1,43 @@
+pin
+---
+
+.. warning::
+
+   Dune Package Management is not final yet and the configuration options
+   are subject to change.
```

Comment:
:art: I would add a pointer to the explanation about package management you wrote. But this is just my trying to create a graph in the documentation :smile: